### PR TITLE
GET endpoints return user. Removed swagger. Moved urls.

### PR DESCRIPTION
**Pull request by oudeismetis**
Connected to #82

### CODE REVIEW

File: franklin/config/urls.py [hunk 1]
```diff
@@ -18,7 +18,6 @@
 
 urlpatterns = [
     url(r'^', include('core.urls')),
-    url(r'^docs/', include('rest_framework_swagger.urls')),
```
Comment (oudeismetis) [inline franklin/config/urls.py]: Swagger was more trouble than it's worth. There was a story to remove it. Several recent changes "broke" the existing docs, so I decided now was a good time to rip this out since the docs are wrong anyway.

File: franklin/core/urls.py [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
 
-from .views import dashboard_init, health
+from .views import health
 from builder.views import UpdateBuildStatus
 from github.views import deploy_hook, deployable_repos, repository_detail, \
         repository_list, get_auth_token
 
 urlpatterns = [
     url(r'^auth/github/$', get_auth_token, name='get_token'),
-    url(r'^dashboard/init/$', dashboard_init, name='dashboard_init'),
```
Comment (oudeismetis) [inline franklin/core/urls.py]: @camiriveron - Instead, you will make a GET call to `auth/github/`. You will still POST to `auth/github` as before.

File: franklin/core/urls.py [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
     url(r'^deployed/$', deploy_hook, name='deploy'),
     url(r'^repos/$', repository_list, name='repo_list'),
     url(r'^repos/(?P<pk>[0-9]+)$', repository_detail, name='repo_details'),
-    url(r'^user/repos/deployable/$', deployable_repos,
-        name='deployable_repos'),
+    url(r'^repos/all/$', deployable_repos, name='deployable_repos'),
```
Comment (oudeismetis) [inline franklin/core/urls.py]: Name seemed odd, so I changed it. @camiriveron

File: franklin/github/api.py [hunk 4]
```diff
@@ -106,23 +104,4 @@ def get_access_token(request):
...
-            }, status=status.HTTP_200_OK)
-        except KeyError:
-            response_data = Response({'status': 'Bad request',
-                                      'message': 'Authentication could not be\
-                                              performed with received data.'},
-                                     status=status.HTTP_400_BAD_REQUEST)
-        return response_data
-    else:
-        return Response(status=r.status_code)
+    return make_rest_post_call(url, headers, params)
```
Comment (oudeismetis) [inline franklin/github/api.py]: Since this file is now a Github SDK, I've moved all app specific logic up into `view.py`.

File: franklin/github/views.py [hunk 5]
```diff
@@ -23,51 +24,19 @@ def repository_list(request):
...
             request.user.details.update_repos_for_user(github_repos)
         sites = request.user.details.sites.filter(is_active=True).all()
-        serializer = SiteSerializer(sites, many=True)
-        return Response(serializer.data)
+        site_serializer = SiteSerializer(sites, many=True)
+        user_serializer = UserSerializer(request.user)
+        return Response({
+            'repos': site_serializer.data,
+            'user': user_serializer.data
+        }, status=status.HTTP_200_OK)
```
Comment (oudeismetis) [inline franklin/github/views.py]: @camiriveron several payloads are now different. Note that the data from the previous response can be found inside `repos`. The `user` object will contain all user data. This will occur on all GET requests that require the passing of an oAuth token. Not currently planning to do this to POST endpoints. We can discuss that though.

File: requirements.txt [hunk 6]
```diff
@@ -7,7 +7,6 @@ djangorestframework==3.2.2
 gunicorn==19.3.0
 pyyaml==3.11
 python-social-auth==0.2.13
-django-rest-swagger==0.3.4
```
Comment (oudeismetis) [inline requirements.txt]: :x:

File: franklin/github/views.py [hunk 7]
```diff
@@ -257,30 +186,27 @@ def deploy_hook(request):
...
+            try:
+                access_token = response.json().get('access_token', None)
+                user = do_auth(access_token)
+                serializer = UserSerializer(user)
+                response_data = Response({
+                    'token': access_token,
+                    'user': serializer.data
+                }, status=status.HTTP_200_OK)
+            except KeyError:
+                response_data = Response(status=status.HTTP_400_BAD_REQUEST)
```
Comment (camiriveron) [inline franklin/github/views.py]: Why did you change this? 

`response_data = Response({'status': 'Bad request',		
 -                                      'message': 'Authentication could not be\		
 -                                              performed with received data.'},		
 -                                     status=status.HTTP_400_BAD_REQUEST)		
 -        return response_data`

Comment (oudeismetis) [inline franklin/github/views.py]: I felt it was redundant. The message was `Bad Request`, but the status code was also `Bad Request`.

**Comment by JulianGindi** [comment]
:+1:
